### ISSUE TITLE
How to pre-split the edges in Link Prediction tasks? 

### ISSUE THREAD

**Issue by robertanto** [snap-stanford/deepsnap#5]
A person may be interested in using a fixed split for link prediction tasks (e.g. given with the dataset). 

In these cases, it may be useful to allow the user to pass a NumPy array containing the list of the edges to use in the training/validation/test set.

**Comment by RexYing** [comment]
Hi, fixed split has been supported by the corresponding argument in pyg_to_graphs if using standard datasets in pytorch geometric. 
For datasets represented by NetworkX, we can simply have 3 graphs (or lists of graphs), corresponding to the train, validation and test set. 
Hope that answers?

**Comment by robertanto** [comment]
@RexYing I looked at the three graphs you mentioned but, looking at the implementation of the split function (for link prediction), is not clear (to me) how to create them. This is mainly due to the implementation of the disjoint message passing during the train.

Personally, I think that a support function which takes as input the list of edges in the 2 (or 3) splits, may be a not negligible contribution to the improvement of the framework.

**Comment by RexYing** [comment]
Hi, there are 2 kinds of splits in this context: disjoint split (which is performed on training set), and the train/val/test split.
For disjoint split, it is always randomly performed (and resampled even), in order to avoid the model overfitting to one particular disjoint split. 
For train/val/test split, this is where you could simply initialize with GraphDataset(train_graphs); GraphDataset(val_graphs) etc. without the need to care about the disjoint split.

So would it help if we isolate an interface to perform disjoint split only (without splitting train, val, test), so that if we have pre-split datasets, we only need to call disjoint split for the training graphs, and directly use GraphDataset(val_graphs) to construct the validation set?

**Comment by robertanto** [comment]
I think it would be great.

Can this function also deal whit the issue #6 (true negative split)?

**Comment by xhe1997** [comment]
function to `pre-split the edges` supported.
